### PR TITLE
selectdata modifications

### PR DESCRIPTION
**Pull request by pantaray**
Author Guidelines
-----------------
- ~[ ] Is the change set **< 400 lines**?~ More like 800
- [X] Was the code checked for memory leaks/performance bottlenecks?
- ~[ ] Is the code running locally **and** on the ESI cluster?~ Only tested locally
- ~[ ] Is the code running on all supported platforms?~ Only tested on Linux

Reviewer Checklist
------------------
- [ ] Are testing routines present?
- [ ] Do **parallel** loops have a set length and correct termination conditions?
- [ ] Do objects in the global package namespace perform proper parsing of their input? 
- [ ] Do code-blocks provide novel functionality, i.e., no re-factoring using builtin/external packages possible?
- [ ] Code layout
  - [ ] Is the code PEP8 compliant?
  - [ ] Does the code adhere to agreed-upon naming conventions?
  - [ ] Are keywords clearly named and easy to understand?
  - [ ] No commented-out code?
- [ ] Are all docstrings complete and accurate?

### CODE REVIEW

File: syncopy/datatype/base_data.py [hunk 1]
```diff
@@ -1931,22 +1952,31 @@ def _selection_setter(self, data, select, dataprop, selectkey):
...
                         steps = np.diff(idxList)
                         if steps.min() == steps.max() == 1:
                             idxList = slice(idxList[0], idxList[-1] + 1, 1)
+
+                    # be careful w/pairwise list-channel selections in `CrossSpectralData` objects
+                    # (that could not be converted to slices above)
+                    if isinstance(idxList, list) and selectkey in ["channel_i", "channel_j"]:
+                        if len(idxList) > 1:
+                            err = "Multi-channel-pair selections not supported"
+                            raise NotImplementedError(err)
```
Comment (tensionhead) [inline syncopy/datatype/base_data.py:1977]: I think this now is a very clear interface, which at the same time reduces the chance of ppl pumping too much stuff into memory :+1:  ..anything more complicated ppl can/should do themselves by e.g. looping over `.show` statements OR preferably by specifying the interesting channels before doing the cross-measure. I'll try to encourage this in the user doc..

File: syncopy/datatype/methods/selectdata.py [hunk 2]
```diff
@@ -17,9 +17,22 @@
 
 @unwrap_cfg
 @detect_parallel_client
-def selectdata(data, trials=None, channels=None, channels_i=None, channels_j=None,
-               toi=None, toilim=None, foi=None, foilim=None, tapers=None, units=None,
-               eventids=None, out=None, inplace=False, clear=False, **kwargs):
+def selectdata(data,
```
Comment (tensionhead) [inline syncopy/datatype/methods/selectdata.py:20]: +1 for readability :)

File: syncopy/datatype/methods/show.py [hunk 3]
```diff
@@ -40,6 +40,10 @@ def show(data, **kwargs):
         determines which keywords can be used.  Some keywords are only valid for
         certain types of Syncopy objects, e.g., "freqs" is not a valid selector
         for an :class:`~syncopy.AnalogData` object.
+    squeeze : bool
```
Comment (tensionhead) [inline syncopy/datatype/methods/show.py:43]: this is great and will hopefully feel quite natural for users (e.g. like numpy behavior)

File: syncopy/nwanalysis/ST_compRoutines.py [hunk 4]
```diff
@@ -191,9 +191,9 @@ class ST_CrossSpectra(ComputationalRoutine):
     def process_metadata(self, data, out):
 
         # Some index gymnastics to get trial begin/end "samples"
-        if data._selection is not None:
-            chanSec = data._selection.channel
-            trl = data._selection.trialdefinition
+        if data.selection is not None:
```
Comment (tensionhead) [inline syncopy/nwanalysis/ST_compRoutines.py:194]: I count on VScode did the heavy lifting here and you didn't have to do this manually like 100's of times :sweat_smile:

File: syncopy/shared/computational_routine.py [hunk 5]
```diff
@@ -906,9 +919,19 @@ def compute_sequential(self, data, out):
...
+                    # (use an explicit `shape` assignment here to avoid copies)
+                    arr.shape = self.sourceShapes[nblock]
+
                     # Perform computation
                     res = self.computeFunction(arr, *argv, **self.cfg)
 
+                    # In case scalar selections have been performed, explicitly assign
+                    # desired output shape to re-create "lost" singleton dimensions
+                    # (use an explicit `shape` assignment here to avoid copies)
+                    res.shape = self.targetShapes[nblock]
```
Comment (tensionhead) [inline syncopy/shared/computational_routine.py:933]: Nice :+1:

**Review by tensionhead — COMMENTED** [review]
Really great work @pantaray ! This will improve the data-interfacing to the users a lot and pave the way for concise plotting and post-processing routines

File: syncopy/datatype/base_data.py (hunk 1 above)
Comment (pantaray) [reply syncopy/datatype/base_data.py:1977]: Good idea! (y)

File: syncopy/nwanalysis/ST_compRoutines.py (hunk 4 above)
Comment (pantaray) [reply syncopy/nwanalysis/ST_compRoutines.py:194]: Nope, luckily VS Code came to the rescue ;)

File: syncopy/datatype/base_data.py [hunk 6]
```diff
@@ -1469,8 +1469,8 @@ def __init__(self, data, select):
                                     varname="select", actual=select)
         if not isinstance(select, dict):
             raise SPYTypeError(select, "select", expected="dict")
-        supported = ["trials", "channels", "channels_i", "channels_j", "toi",
-                     "toilim", "foi", "foilim", "tapers", "units", "eventids"]
+        supported = ["trials", "channel", "channel_i", "channel_j", "toi",
```
[the remark this answers is not in the document]
Comment (tensionhead) [reply syncopy/datatype/base_data.py:1472]: Yap that would remove some of the clutter right? In the end the doc will have to address this, most important thing is to capture stuff like `channels_is` which we are doing now :+1:

[the remark this answers is not in the document]
Comment (pantaray) [reply syncopy/datatype/base_data.py:1472]: OK, I'll refactor this part accordingly then. PR commit incoming...

**Pull request merged**